lineapy/api/models/linea_artifact.py

```diff
@@ -105,36 +103,118 @@ def get_value(self) -> object:
         """
         Get and return the value of the artifact
         """
-        pickle_filename = self.db.get_node_value_path(
-            self._node_id, self._execution_id
-        )
-        if pickle_filename is None:
+        metadata = self.get_metadata()
+        linea_metadata = metadata["lineapy"]
+        saved_filepath = linea_metadata.storage_path
+        if saved_filepath is None:
             return None
         else:
-            # TODO - set unicode etc here
             track(GetValueEvent(has_value=True))
+            # read from mlflow
+            if "mlflow" in metadata.keys():
+                return read_mlflow(metadata["mlflow"])
+
+            # read from lineapy
+            return self._read_pickle(saved_filepath)
+
+    @lru_cache(maxsize=None)
+    def _get_storage_path(self) -> Optional[str]:
+        return self.db.get_node_value_path(self._node_id, self._execution_id)
+
+    @lru_cache(maxsize=None)
+    def get_metadata(self, lineapy_only: bool = False) -> ArtifactInfo:
+        """
+        Get artifact backend storage metadata
+
+        :param lineapy_only: If ``False``, will include both LineaPy related
+            metadata and metadata from storage backend(if it is not LineaPy).
+            If ``True``, will only return LineaPy related metadata no matter
+            which storage backend is using.
+
+        :return: Metadata for artifact backend storage.
+
+        """
 
-            artifact_storage_dir = options.safe_get("artifact_storage_dir")
-            filepath = (
-                artifact_storage_dir.joinpath(pickle_filename)
-                if isinstance(artifact_storage_dir, Path)
-                else f'{artifact_storage_dir.rstrip("/")}/{pickle_filename}'
+        if self._artifact_id is None or self.date_created is None:
```

Comment:
This logic seems like it would be good in an init or post_init to populate 
`_artifact_id` and `date_created`.